bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/dialogs/SearchPattern.java

```diff
@@ -535,15 +616,13 @@ protected boolean isNameCharAllowed(char nameChar) {
 	}
 
 	/**
-	 * Returns the rule to apply for matching keys. Can be exact match, prefix
-	 * match, pattern match or camelcase match. Rule can also be combined with a
-	 * case sensitivity flag.
-	 *
-	 * @return one of RULE_EXACT_MATCH, RULE_PREFIX_MATCH, RULE_PATTERN_MATCH,
-	 *         RULE_CAMELCASE_MATCH, combined with RULE_CASE_SENSITIVE, e.g.
-	 *         RULE_EXACT_MATCH | RULE_CASE_SENSITIVE if an exact and case sensitive
-	 *         match is requested, or RULE_PREFIX_MATCH if a prefix non case
-	 *         sensitive match is requested.
+	 * Returns the active rule to apply for matching keys, based on the currently
+	 * set {@link #setPattern(String) pattern} and allowed rules passed to the
+	 * {@link #SearchPattern(int) constructor}.
+	 *
+	 * @return one of {@link #RULE_BLANK_MATCH}, {@link #RULE_EXACT_MATCH},
+	 *         {@link #RULE_PREFIX_MATCH}, {@link #RULE_PATTERN_MATCH},
+	 *         {@link #RULE_CAMELCASE_MATCH}
```

Comment:
Well, if I haven't overlooked st, this is not a "breaking change", but just "fixing the javadoc" - to be unambiguous & complete (which I have somewhat mentioned in a previous discussion, if I remember correctly).

So, 1) `RULE_BLANK_MATCH` should be mentioned to make it complete and 2) "case sensitivity flag and how it can be combined" is already described in the linked constructor's javadoc & it is not quite relevant to what this method returns, therefore removing this part from here makes perfect sense to me.

Reply:
Could those points be discussed in a separate PR?
Fixing (code or javadoc or whatever) is always welcome, with highest priority; so we don't want it to be a "by the way" thing. Keeping things separate allows more efficient progress and also makes development safer: it's easy to revert a new feature without reverting the "by the way" fixes, so fixes are less likely to disappear.

Reply:
Good point, I will try to move all the javadoc-related fixes into a separate PR, OK? I can see now there are way too many problems from the past (little to no code review there?), so mixing their fix into this PR is problematic...

Reply:
=> #469